### PR TITLE
feat(planner/python): add sanic support

### PR DESCRIPTION
**Pull request by CorrectRoadH**
[Sanic](https://sanic.dev/zh/) is a Python 3.8+ web server and web framework.

### CODE REVIEW

File: internal/python/plan.go [hunk 1]
```diff
@@ -261,6 +266,24 @@ func DetermineWsgi(ctx *pythonPlanContext) string {
...
+
+		re := regexp.MustCompile(`(\w+)\s*=\s*Sanic\([^)]*\)`)
+		content, err := afero.ReadFile(src, entryFile)
+		if err != nil {
+			return ""
+		}
+
+		match := re.FindStringSubmatch(string(content))
+		if len(match) > 1 {
+			entryWithoutExt := strings.Replace(entryFile, ".py", "", 1)
```
Comment (pan93412) [inline internal/python/plan.go:280]:
```suggestion
			entryWithoutExt := strings.TrimSuffix(entryFile, ".py")
```

TrimSuffix is more efficient than Replace.

File: internal/python/plan.go [hunk 2]
```diff
@@ -535,6 +558,9 @@ func determineStartCmd(ctx *pythonPlanContext) string {
 
 		if framework == types.PythonFrameworkFastapi {
 			commandSegment = append(commandSegment, "uvicorn", wsgi, "--host 0.0.0.0", "--port "+wsgilistenedPort)
+		}
+		if framework == types.PythonFrameworkSanic {
```
Comment (pan93412) [inline internal/python/plan.go:562]: Use `else if` is better.

```suggestion
		else if framework == types.PythonFrameworkSanic {
```

File: tests/python-sanic/__pycache__/server.cpython-38.pyc
Comment (pan93412) [inline tests/python-sanic/__pycache__/server.cpython-38.pyc:1]: Do not commit `__pycache__` to tests.

**Review by pan93412 — CHANGES_REQUESTED** [review]

**Comment by CorrectRoadH** [comment]
@pan93412 Please take a look

**Review by pan93412 — APPROVED** [review]

**Review by yuaanlin — APPROVED** [review]

**Pull request merged**